MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectJsonMenuComponent.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Components/UI/UiGameLiftConnectJsonMenuComponent.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+
+#include <LyShine/Bus/UiButtonBus.h>
+#include <LyShine/Bus/UiCursorBus.h>
+#include <LyShine/Bus/UiElementBus.h>
+#include <LyShine/Bus/UiTextBus.h>
+#include <LyShine/Bus/UiTextInputBus.h>
+#include <LyShine/Bus/UiInteractableBus.h>
+
+#include <Multiplayer/Session/SessionRequests.h>
+#include <Request/AWSGameLiftSessionRequestBus.h>
+
+
+namespace MPSGameLift
+{
+    void UiGameLiftConnectJsonMenuComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (const auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<UiGameLiftConnectJsonMenuComponent, AZ::Component>()
+                ->Version(1)
+                ->Field("ConnectButton", &UiGameLiftConnectJsonMenuComponent::m_connectButtonUi)
+                ->Field("ExitButton", &UiGameLiftConnectJsonMenuComponent::m_quitButtonUi)
+                ->Field("JsonInput", &UiGameLiftConnectJsonMenuComponent::m_jsonInputUi)
```

Comment:
can we name this for what the JSON contains? I.e., "PlayerSessionJsonInput"

Reply:
Absolutely, renamed to PlayerSessionDataInputUi